### ISSUE TITLE
Feature Request: Ask Permission before Downloading (with option to save file temporarily) 

### ISSUE THREAD

**Issue by MintMain21** [inbasic/turbo-download-manager-v2#35]
In Firefox 98, the default behavior for downloading files changed. Before, it would ask if you wanted to save the file permanently, open it for temporary use, or cancel/block the download altogether. Now, it automatically begins the download, with the Downloads folder as the default location, with the only option available being to cancel the download altogether. I haven't used Google Chrome since 2019, but this seems to be in line with how Downloads are handled there.

Seeing as this is a cross-platform extension, it may be handy to emulate these options within the extension in the event there is no native way to create this in Chrome or Firefox moving forward. 

TL;DR, I propose the following options to be added to the extension.

1. The option to ask before downloading anything. 
2. The option to choose download locations, including to temporary folders.
